instant-xml-macros/src/de.rs

```diff
@@ -0,0 +1,271 @@
+use proc_macro2::{Ident, Span, TokenStream};
+use quote::quote;
+
+use crate::{get_namespaces, retrieve_attr};
+
+struct Tokens<'a> {
+    enum_: &'a mut TokenStream,
+    consts_: &'a mut TokenStream,
+    names_: &'a mut TokenStream,
+    match_: &'a mut TokenStream,
+}
+
+pub struct Deserializer {
+    fn_vec: Vec<TokenStream>,
+}
+
+impl Deserializer {
+    pub fn new(input: &syn::DeriveInput) -> Deserializer {
+        let ident = &input.ident;
+        let name = ident.to_string();
+        let mut fn_vec = Vec::new();
+
+        let (_, other_namespaces) = get_namespaces(&input.attrs);
+        let mut namespaces_map: TokenStream = proc_macro::TokenStream::from(
+            quote!(let mut namespaces_map = std::collections::HashMap::new();),
+        )
+        .into();
+        for (k, v) in other_namespaces.iter() {
+            namespaces_map.extend(quote!(
+                namespaces_map.insert(#k, #v);
+            ))
+        }
+
+        // Elements
+        let mut elements_enum = TokenStream::new();
+        let mut elements_lets = TokenStream::new();
+        let mut elements_names = TokenStream::new();
+        let mut elem_type_match = TokenStream::new();
+        let mut elements_tokens = Tokens {
+            enum_: &mut elements_enum,
+            consts_: &mut elements_lets,
+            names_: &mut elements_names,
+            match_: &mut elem_type_match,
+        };
+
+        // Attributes
+        let mut attributes_enum = TokenStream::new();
+        let mut attributes_lets = TokenStream::new();
+        let mut attributes_names = TokenStream::new();
+        let mut attr_type_match = TokenStream::new();
+        let mut attributes_tokens = Tokens {
+            enum_: &mut attributes_enum,
+            consts_: &mut attributes_lets,
+            names_: &mut attributes_names,
+            match_: &mut attr_type_match,
+        };
+
+        // Common values
+        let mut declare_values = TokenStream::new();
+        let mut return_val = TokenStream::new();
+
+        match &input.data {
+            syn::Data::Struct(ref data) => {
+                match data.fields {
+                    syn::Fields::Named(ref fields) => {
+                        fields.named.iter().enumerate().for_each(|(index, field)| {
+                            if let Some(true) = retrieve_attr("attribute", &field.attrs) {
+                                Self::process_field(
```

Comment:
If the only difference between the two calls to `process_field()` here is the 3rd and 6th arguments, we should write the code so that this is obvious.

Reply:
I changed the arguments order so now the last ones varies. I hope this is what you meant.

Reply:
No, I meant that there should only be one call to `Self::process_field()`. You can then switch the arguments around where the argument values need to be different using an `if`.

Reply:
I unified it, but it requires some overhead, if we are deciding inside of function, we don't know which `tokens` we need to append to. Please take a look at this commit:
https://github.com/InstantDomain/instant-xml/pull/7/commits/5966cde0a06b14ed7619c9ba1dca9164436a691f

Reply:
done